crates/menmos-client/src/builder.rs

```diff
@@ -74,6 +76,11 @@ impl ClientBuilder {
         self
     }
 
+    pub fn with_smart_detection(mut self) -> Self {
```

Comment:
Really a nitpick, but since we renamed the detector to `MetadataDetector` I'd name this `with_metadata_detection`

Reply:
Good catch! I simply missed it.